lib/IRGen/GenStruct.cpp

```diff
@@ -351,6 +358,73 @@ namespace {
                                   ClangFieldInfo> {
     const clang::RecordDecl *ClangDecl;
 
+    const clang::CXXConstructorDecl *findCopyConstructor() const {
+      const clang::CXXRecordDecl *cxxRecordDecl =
+          dyn_cast<clang::CXXRecordDecl>(ClangDecl);
+      if (!cxxRecordDecl)
+        return nullptr;
+      for (auto method : cxxRecordDecl->methods()) {
+        if (auto ctor = dyn_cast<clang::CXXConstructorDecl>(method)) {
+          if (ctor->isCopyConstructor())
+            return ctor;
+        }
+      }
+      return nullptr;
+    }
+
+    CanSILFunctionType createCXXCopyConstructorFunctionType(IRGenFunction &IGF,
+                                                            SILType T) const {
+      // Create the following function type:
+      //   @convention(c) (UnsafePointer<T>) -> @out T
+      // This is how clang *would* import the copy constructor. So, later, when
+      // we pass it to "emitCXXConstructorThunkIfNeeded" we get a thunk with
+      // the following LLVM function type:
+      //   void (%struct.T* %this, %struct.T* %0)
+      auto ptrTypeDecl =
+          IGF.getSILModule().getASTContext().getUnsafePointerDecl();
+      auto subst = SubstitutionMap::get(ptrTypeDecl->getGenericSignature(),
+                                        {T.getASTType()},
+                                        ArrayRef<ProtocolConformanceRef>{});
+      auto ptrType = ptrTypeDecl->getDeclaredInterfaceType().subst(subst);
+      SILParameterInfo ptrParam(ptrType->getCanonicalType(),
+                                ParameterConvention::Direct_Unowned);
+      SILResultInfo result(T.getASTType(), ResultConvention::Indirect);
+
+      return SILFunctionType::get(
+          GenericSignature(),
+          SILFunctionType::ExtInfo().withRepresentation(
+              SILFunctionTypeRepresentation::CFunctionPointer),
+          SILCoroutineKind::None,
+          /*callee*/ ParameterConvention::Direct_Unowned,
```

Comment:
Please use the following comment format:

```
/*callee=*/ ParameterConvention::Direct_Unowned
```

Comments in this style are automatically checked by ClangTidy against the function definition ensuring that the text in the comment matches the parameter name.

Reply:
Never knew that -- thanks!